File: call-management/rule-management/create-company-greeting/code-samples/createCompanyGreeting.py
# https://developers.ringcentral.com/my-account.html#/applications
# Find your credentials at the above url, set them as environment variables, or enter them below

# PATH PARAMETERS
accountId = '<ENTER VALUE>'

import os
from ringcentral import SDK
rcsdk = SDK(os.environ['clientId'], os.environ['clientSecret'], os.environ['serverURL'])
platform = rcsdk.platform()
platform.login(os.environ['username'], os.environ['extension'], os.environ['password'])

builder = rcsdk.create_multipart_builder()
builder.set_body({
    'type': 'Company'
})

binary = ('mygreeting.wav', open('mygreeting.wav','r').read(), 'audio/wav')
builder.add(binary)

request = builder.request(f'/restapi/v1.0/account/{accountId}/greeting')

resp = platform.send_request(request)

File: call-management/rule-management/create-company-greeting/code-samples/createCompanyGreeting.php
<?php
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
$accountId = '<ENTER VALUE>';

require('vendor/autoload.php');
$rcsdk = new RingCentral\SDK\SDK(getenv('clientId'), getenv('clientSecret'), getenv('serverURL'));
$platform = $rcsdk->platform();
$platform->login(getenv('username'), getenv('extension'), getenv('password'));

$request = $rcsdk->createMultipartBuilder()
    ->setBody(array(
        'type' => 'Company'
    ))
    ->add('binary', 'mygreeting.wav');
    ->add(fopen('./mygreeting.wav', 'r'))
    ->request("/restapi/v1.0/account/{$accountId}/greeting");

$r = $platform->sendRequest($request);
?>
